cli/src/pcluster/validators/scheduler_plugin_validators.py

```diff
@@ -22,3 +29,82 @@ def _validate(self, grant_sudo_privileges: bool, requires_sudo_privileges: bool)
                 "but these privileges were not granted because GrantSudoPrivileges was not set or set to false.",
                 FailureLevel.ERROR,
             )
+
+
+class SchedulerPluginOsArchitectureValidator(Validator):
+    """Verify that head node architecture and os combination is supported by the scheduler plugin."""
+
+    def _validate(self, os, architecture: str, supported_x86, supported_arm64):
+        oss_and_supported_architectures_mapping = {"x86_64": supported_x86, "arm64": supported_arm64}
+        supported_oss = oss_and_supported_architectures_mapping.get(architecture)
+        if os not in supported_oss:
+            self._add_failure(
+                "The scheduler plugin supports the OSs {0} for architecture {1}, none of which are "
+                "compatible with the configured OS ({2}) and the architecture supported by the head node instance "
+                "type ({1}).".format(supported_oss, architecture, os),
+                FailureLevel.ERROR,
+            )
+
+
+class SchedulerPluginRegionValidator(Validator):
+    """Verify that the region is supported by the scheduler plugin."""
+
+    def _validate(self, region, supported_regions):
+        if supported_regions and region not in supported_regions:
+            self._add_failure(
+                "The specified region {0} is not supported by the scheduler plugin. "
+                "Supported regions are: {1}.".format(region, supported_regions),
+                FailureLevel.ERROR,
+            )
+
+
+class SupportedVersionsValidator(Validator):
+    """
+    Validate if the installed ParallelCluster Version is supported by the scheduler plugin.
+
+    The supported formats for SupportedParallelClusterVersions is either a string contains list of versions or a range.
+    Example: "3.0.0, 3.0.1" or ">=3.1.0, <=3.1.0"
+    """
+
+    def _validate(self, installed_version, supported_versions_string):
+        if supported_versions_string:
+            comparison_operators = {
+                "<": operator.lt,
+                "<=": operator.le,
+                "==": operator.eq,
+                "!=": operator.ne,
+                ">=": operator.ge,
+                ">": operator.gt,
+            }
+            supported_versions_list = [version.strip() for version in supported_versions_string.split(",")]
+            if any(x in supported_versions_string for x in [">", "<"]):  # Input string is a range
+                requirements = []
+                for version in supported_versions_list:
+                    try:
+                        match = re.search(r"^([<>=!]+)([\d.]+)", version)
+                        requirements.append(CliRequirement(operator=match.group(1), version=match.group(2)))
+                    except Exception as e:
+                        self._add_failure(
+                            "Unable to parse SupportedParallelClusterVersions {0} in the scheduler plugin: {1}.".format(
```

Comment:
let's give a hint on the supported values/format in the error message.

Reply:
Done